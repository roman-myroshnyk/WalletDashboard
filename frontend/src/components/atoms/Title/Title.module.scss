@import '@/styles/colors.scss';

.title {
  color: $white;
  font-weight: 800;
  font-size: 26px;
  font-family: Epilogue, sans-serif;
  font-style: normal;
  line-height: 26px;
  text-align: center;
  text-decoration: none;
}
