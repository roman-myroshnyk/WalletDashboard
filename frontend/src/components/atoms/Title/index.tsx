import { FC, ReactNode } from 'react';
import styles from './Title.module.scss';

export interface IProps {
    children: ReactNode;
}

const Title:FC<IProps> = ({ children }) => (
  <h3 className={styles.title}>
    {children}
  </h3>
);

export default Title;
