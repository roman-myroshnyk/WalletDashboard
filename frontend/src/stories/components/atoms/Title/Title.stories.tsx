import { Meta, Story } from '@storybook/react/types-6-0';
import Title, { IProps } from '@/atoms/Title';

export default {
  title: 'atoms/Title',
  component: Title,
  argTypes: {
    text: {
      controls: 'text',
    },
  },
} as Meta;

interface ITemplateProps extends IProps {
  text: string;
}

const Template:Story<ITemplateProps> = ({ ...args }) => (
  <Title {...args}>
    {args.text}
  </Title>
);

export const Default = Template.bind({});

Default.args = {
  text: 'Digital Wallet Dashboard',
};
