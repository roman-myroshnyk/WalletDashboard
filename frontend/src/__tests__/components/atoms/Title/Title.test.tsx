import '.../../../mocks/framer-motion';
import { render } from '@testing-library/react';
import Title from '@/atoms/Title';

describe('atoms/Title', () => {
  it('Title should render', () => {
    const { container } = render(
      <Title>
        My title
      </Title>,
    );
    expect(container).toMatchSnapshot();
  });
});
