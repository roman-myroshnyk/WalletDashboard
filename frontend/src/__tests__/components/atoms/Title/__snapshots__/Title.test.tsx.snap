// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/Title Title should render 1`] = `
<div>
  <h3
    class="title"
  >
    My title
  </h3>
</div>
`;
